pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/ShadowReplicator.java

```diff
@@ -113,6 +117,7 @@ protected boolean replicateEntries(List<Entry> entries) {
 
                 // Increment pending messages for messages produced locally
                 PENDING_MESSAGES_UPDATER.incrementAndGet(this);
+                lastSent = entry.getPosition();
```

Comment:
Same as the comment in GeoReplicator.

Reply:
The reply for this comment is the same as https://github.com/apache/pulsar/pull/20128#discussion_r1171246436